dynakube.go

```diff
@@ -112,7 +116,36 @@ func (c *Client) Get(ctx context.Context, key client.ObjectKey, obj client.Objec
 
 // List retrieves list of objects for a given namespace and list options.
 func (c *Client) List(ctx context.Context, list client.ObjectList, opts ...client.ListOption) error {
-	panic("implement me")
+	listOpts := client.ListOptions{}
+	listOpts.ApplyOptions(opts)
+	metav1ListOpts := listOpts.AsListOptions()
+	if metav1ListOpts == nil {
+		return fmt.Errorf("unable to convert %T to %T", listOpts, metav1ListOpts)
+	}
+
+	gvk, _ := apiutil.GVKForObject(list, c.scheme)
+	gvk.Kind = gvk.Kind[:len(gvk.Kind)-4] /*base library appends List*/
```

Comment:
this is wild ...

Reply:
you'll never guess where I got it from: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/dynamic/fake/simple.go#L356